packages/jsts/src/rules/S6844/rule.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * SonarQube JavaScript Plugin
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+// https://sonarsource.github.io/rspec/#/rspec/S6844/javascript
+
+import { rules as jsxA11yRules } from 'eslint-plugin-jsx-a11y';
+import { interceptReport } from '../helpers';
+
+const anchorIsValid = jsxA11yRules['anchor-is-valid'];
+
+export const rule = interceptReport(anchorIsValid, (context, reportDescriptor) => {
+  const msg = (reportDescriptor as any)!.message;
+  (reportDescriptor as any)!.message = msg.substring(0, msg.indexOf(' Learn'));
+  context.report(reportDescriptor);
```

Comment:
```suggestion
  const descriptor = reportDescriptor as any;
  const { message } = descriptor;
  descriptor.message = message.substring(0, message.indexOf(' Learn'));
  context.report(descriptor);
```

Reply:
Another suggestion using a regex, but I am not sure it's any better than calling `substring` and `indexOf`:

```suggestion
  const descriptor = reportDescriptor as any;
  const { message } = descriptor;
  descriptor.message = message.replace(/ Learn.*/, '');
  context.report(descriptor);
```

Reply:
In case you ignore my first two suggestions, we can at least get rid of the non-null assertion operator `(!)` since we cast to `any` anyway:

```suggestion
  const msg = (reportDescriptor as any).message;
  (reportDescriptor as any).message = msg.substring(0, msg.indexOf(' Learn'));
  context.report(reportDescriptor);
```